fs/zbd_zenfs.h

```diff
@@ -79,6 +81,8 @@ class ZonedBlockDevice {
   std::atomic<long> active_io_zones_;
   std::atomic<long> open_io_zones_;
   std::condition_variable zone_resources_;
+  std::mutex zone_close_mutex_;
+  boost::optional<IOStatus> zone_close_error_;
```

Comment:
Let's make this "zone_close_error_" a generic "zone_management_status" in stead so we can use it for other stuff that goes wrong in the future. zone_management_status should be initialized to ok.